File: 104. Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.height(root, 0)
    
    def height(self, root, h):
        if not root:
            return h
        return max(self.height(root.left, h), self.height(root.right, h)) + 1

File: 104. Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        return height(root, 0);
    }
    
    int height(TreeNode* root, int h)
    {
        if (root == NULL) return h;
        int l = height(root->left, h), r = height(root->right, h);
        return l > r ? l + 1 : r + 1;
    }
};
